tracing-appender/src/lib.rs

```diff
@@ -42,7 +107,61 @@ mod worker;
 /// Unintentional drops of `WorkerGuard` remove the guarantee that logs will be flushed
 /// during a program's termination, in a panic or otherwise.
 ///
+/// See [`WorkerGuard`][worker_guard] for examples of using the guard.
+///
+/// [worker_guard]: ./struct.WorkerGuard.html
+///
 /// # Examples
+///
+/// ``` rust
+/// # fn docs() {
+/// let (non_blocking, _guard) = tracing_appender::non_blocking(std::io::stdout());
+/// let subscriber = tracing_subscriber::fmt().with_writer(non_blocking);
+/// tracing::subscriber::with_default(subscriber.finish(), || {
+///    tracing::event!(tracing::Level::INFO, "Hello");
+/// });
+/// # }
+/// ```
+pub mod non_blocking;
```

Comment:
In general, I prefer to put module level docs _inside_ the module they document. But, this isn't a blocker, so take it or leave it.